src/Compute/v2/Models/QuotaSet.php

```diff
@@ -0,0 +1,170 @@
+<?php declare(strict_types=1);
+
+namespace OpenStack\Compute\v2\Models;
+
+use OpenStack\Common\Resource\Deletable;
+use OpenStack\Common\Resource\OperatorResource;
+use OpenStack\Common\Resource\Retrievable;
+use OpenStack\Common\Resource\Updateable;
+
+/**
+ * Represents a Compute v2 Quota Set
+ *
+ * @property \OpenStack\Compute\v2\Api $api
+ */
+class QuotaSet extends OperatorResource implements Retrievable, Updateable, Deletable
+{
+    /**
+     * The number of allowed instance cores for each tenant.
+     *
+     * @var  int|array
```

Comment:
Exactly, but then this is how Openstack implemented it. 

For `os-quota-sets/{tenantId}` returns `int`; `os-quota-sets/{tenantId}/detail` returns an array.  See https://developer.openstack.org/api-ref/compute/?expanded=show-the-detail-of-quota-detail#show-the-detail-of-quota